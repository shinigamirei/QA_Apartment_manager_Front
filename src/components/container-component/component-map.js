import { AdminHome, Admin1, Admin2, Admin3, Admin4, Admin5, Admin6,
         RecruiterHome, Recruiter1, Recruiter2, Recruiter3, Recruiter4, Recruiter5, Recruiter6 } from'./placeholder-components';
import Login from './login.component'
import React, { Component }  from 'react';
import ApartmentLanding from '../apartment_manager_landing.component';




let func = function(x){console.log(x)}


export const myConfig =
{
    "null": <Login/>,
    "recruiter": {
        "home": <RecruiterHome/>,
        "buttons": [
                        {'name': 'Home', 'content': <RecruiterHome/>}
                   ],
        "dropdowns":[
                        {   'name': 'Bursary',
                            'content':[
                                        {'name': 'Recruiter Bursary Link 1', 'content': <Recruiter1/>},
                                        {'name': 'Recruiter Bursary Link 2', 'content': <Recruiter2/>}
                                    ]
                        },
                        {  'name': 'Apartments',
                           'content':[
                                        {'name': 'Recruiter Apartment Link 1', 'content': <Recruiter3/>},
                                        {'name': 'Recruiter Apartment Link 2', 'content': <Recruiter4/>}
                                    ]
                        }
                    ],
        "side": [
                        {'name': 'Recruiter SideBar Link 1', 'content': <Recruiter5/>},
                        {'name': 'Recruiter SiderBar Link 2', 'content': <Recruiter6/>}
                   ],
    },
    "admin": {
        "home": <ApartmentLanding className="ApartmentLandingComponent"></ApartmentLanding>,
        "buttons": [
                        {'name': 'Home', 'content': <ApartmentLanding className="ApartmentLandingComponent"></ApartmentLanding>}
                   ],
        "dropdowns":[
                        {   'name': 'Demo Dropdown',
                            'content':[
                                        {'name': 'Admin placeholder 1', 'content':<AdminHome id="helloworld"/>},
                                        {'name': 'Admin placeholder 2', 'content':<Admin1/>},
                                        {'name': 'Admin placeholder 3', 'content': <Admin2/>},
                                        {'name': 'Admin placeholder 4', 'content': <Admin3/>},
                                        {'name': 'Admin placeholder 5', 'content': <Admin4/>}
                                      ]
                        }
                    ],
        "side": [
                        {'name': 'Admin Sidebar Link 1', 'content': <Admin5/>},
                        {'name': 'Admin Sidebar Link 2', 'content': <Admin6/>}
                   ],
    }
}