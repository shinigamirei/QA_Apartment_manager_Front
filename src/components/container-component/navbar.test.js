import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow, mount, render, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import MockAdapter from 'axios-mock-adapter';
import renderer from 'react-test-renderer';
import Navbarr from './navbar.component';
import { myConfig } from './component-map';

configure({ adapter: new Adapter() });

describe('Navbar', () => {

    it('should render the navbar with admin links when role is admin', () => {
        const testRole = 'admin';

        const component = shallow(<Navbarr role={testRole} links={myConfig}/>);

        expect(component).toMatchSnapshot();
    });

    it('should render the navbar with recruiter links when role is recruiter', () => {
        const testRole = 'recruiter';

        const component = shallow(<Navbarr role={testRole} links={myConfig}/>);

        expect(component).toMatchSnapshot();
    });

   
});