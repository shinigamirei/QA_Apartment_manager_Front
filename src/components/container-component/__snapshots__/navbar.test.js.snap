// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render the navbar with admin links when role is admin 1`] = `
<div
  id="navigation-bar"
>
  <Bootstrap(undefined)
    bg="light"
    expand="lg"
  >
    <Nav
      className="mr-auto"
      fill={false}
      justify={false}
    >
      <NavLink
        as={[Function]}
        disabled={false}
        onClick={[Function]}
      >
        Home
      </NavLink>
      <NavDropdown
        id="basic-nav-dropdown"
        title="Component Demos"
      >
        <DropdownItem
          as={[Function]}
          disabled={false}
          onClick={[Function]}
        >
          Table
        </DropdownItem>
        <DropdownItem
          as={[Function]}
          disabled={false}
          onClick={[Function]}
        >
          Search Bar
        </DropdownItem>
        <DropdownItem
          as={[Function]}
          disabled={false}
          onClick={[Function]}
        >
          Buttons
        </DropdownItem>
        <DropdownItem
          as={[Function]}
          disabled={false}
          onClick={[Function]}
        >
          Apartment Display
        </DropdownItem>
        <DropdownItem
          as={[Function]}
          disabled={false}
          onClick={[Function]}
        >
          Calendar
        </DropdownItem>
        <DropdownItem
          as={[Function]}
          disabled={false}
          onClick={[Function]}
        >
          Generated Form
        </DropdownItem>
      </NavDropdown>
    </Nav>
  </Bootstrap(undefined)>
</div>
`;

exports[`Navbar should render the navbar with recruiter links when role is recruiter 1`] = `
<div
  id="navigation-bar"
>
  <Bootstrap(undefined)
    bg="light"
    expand="lg"
  >
    <Nav
      className="mr-auto"
      fill={false}
      justify={false}
    >
      <NavLink
        as={[Function]}
        disabled={false}
        onClick={[Function]}
      >
        Home
      </NavLink>
      <NavDropdown
        id="basic-nav-dropdown"
        title="Bursary"
      >
        <DropdownItem
          as={[Function]}
          disabled={false}
          onClick={[Function]}
        >
          Recruiter Bursary Link 1
        </DropdownItem>
        <DropdownItem
          as={[Function]}
          disabled={false}
          onClick={[Function]}
        >
          Recruiter Bursary Link 2
        </DropdownItem>
      </NavDropdown>
      <NavDropdown
        id="basic-nav-dropdown"
        title="Apartments"
      >
        <DropdownItem
          as={[Function]}
          disabled={false}
          onClick={[Function]}
        >
          Recruiter Apartment Link 1
        </DropdownItem>
        <DropdownItem
          as={[Function]}
          disabled={false}
          onClick={[Function]}
        >
          Recruiter Apartment Link 2
        </DropdownItem>
      </NavDropdown>
    </Nav>
  </Bootstrap(undefined)>
</div>
`;
